internal/services/datafactory/data_factory.go

```diff
@@ -144,6 +169,39 @@ func flattenDataFactoryVariables(input map[string]*datafactory.VariableSpecifica
 	return output
 }
 
+func expandDataFactoryVariablesFourPointOh(input []interface{}) map[string]*datafactory.VariableSpecification {
+	variables := make(map[string]*datafactory.VariableSpecification)
+	for _, v := range input {
+		val := v.(map[string]interface{})
+
+		variables[val["name"].(string)] = &datafactory.VariableSpecification{
+			Type:         datafactory.VariableType(val["type"].(string)),
+			DefaultValue: val["default_value"],
+		}
+	}
+	return variables
+}
+
+func flattenDataFactoryVariablesFourPointOh(input map[string]*datafactory.VariableSpecification) []interface{} {
+	variables := make([]interface{}, 0, len(input))
+	for k, v := range input {
+
+		// convert value to string if it is bool
```

Comment:
Is this only necessary for variable? How about parameters?

Reply:
the api can do the conversion for parameters but not for variables